core/src/main/java/org/apache/druid/data/input/impl/MapInputRowParser.java

```diff
@@ -36,27 +36,28 @@
 public class MapInputRowParser implements InputRowParser<Map<String, Object>>
 {
   private final ParseSpec parseSpec;
+  private final List<String> dimensions;
 
   @JsonCreator
   public MapInputRowParser(
       @JsonProperty("parseSpec") ParseSpec parseSpec
   )
   {
     this.parseSpec = parseSpec;
+    this.dimensions = parseSpec.getDimensionsSpec().getDimensionNames();
   }
 
   @Override
   public List<InputRow> parseBatch(Map<String, Object> theMap)
   {
-    final List<String> dimensions = parseSpec.getDimensionsSpec().hasCustomDimensions()
-                                    ? parseSpec.getDimensionsSpec().getDimensionNames()
-                                    : Lists.newArrayList(
-                                        Sets.difference(
-                                            theMap.keySet(),
-                                            parseSpec.getDimensionsSpec()
-                                                     .getDimensionExclusions()
-                                        )
-                                    );
+    final List<String> dimensions;
+    if (!this.dimensions.isEmpty()) {
+      dimensions = this.dimensions;
+    } else {
+      dimensions = Lists.newArrayList(
```

Comment:
Wasn't `Lists.newArrayList` prohibited recently by #6511?

Reply:
No, #6511 prohibite `Lists#newArrayList()`, `Lists.newArrayList#newArrayList(java.lang.Iterable)` has not been prohibited because there is no JDK apis to replace it.

Reply:
Got it, I didn't look too closely. Perhaps I don't quite understand the point of the prohibition, but it makes me wonder if we should prefer to use `new ArrayList<>(...` when possible or is there some reason not to do that here since `Sets.difference` is returning a `Collection`? I seriously don't care either way, just wondering what makes the most sense going forward.

Reply:
No reason, I just didn't realize it can be modified to `new ArrayList<>(...`. In #6588, you have modified it and it's fine for me. The point of the prohibition is to reduce dependency on guava and use JDK collection APIs instead.

Reply:
Oh, I reverted it in #6588, was an accidental change in my working copy i did while looking at this PR to see if it could be done heh :)